src/Proto.Actor/Futures.cs

```diff
@@ -1,33 +1,39 @@
-﻿// -----------------------------------------------------------------------
-//  <copyright file="Futures.cs" company="Asynkron HB">
-//      Copyright (C) 2015-2017 Asynkron HB All rights reserved
-//  </copyright>
-// -----------------------------------------------------------------------
-
-using System.Threading.Tasks;
-
-namespace Proto
-{
-    public class FutureProcess<T> : Process
-    {
-        private readonly TaskCompletionSource<T> _tcs;
-
-        public FutureProcess(TaskCompletionSource<T> tcs)
-        {
-            _tcs = tcs;
-        }
-
-        public override void SendUserMessage(PID pid, object message, PID sender)
-        {
-            if (message is T)
-            {
-                _tcs.TrySetResult((T) message);
-                pid.Stop();
-            }
-        }
-
-        public override void SendSystemMessage(PID pid, object message)
-        {
-        }
-    }
+﻿// -----------------------------------------------------------------------
+//  <copyright file="Futures.cs" company="Asynkron HB">
+//      Copyright (C) 2015-2017 Asynkron HB All rights reserved
+//  </copyright>
+// -----------------------------------------------------------------------
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Proto
+{
+    public class FutureProcess<T> : Process
+    {
+        private readonly TaskCompletionSource<T> _tcs;
+        private readonly CancellationToken _cancellationToken;
+
+        public FutureProcess(TaskCompletionSource<T> tcs) : this(tcs, CancellationToken.None) { }
+        public FutureProcess(TaskCompletionSource<T> tcs, CancellationToken cancellationToken)
+        {
+            _tcs = tcs;
+            _cancellationToken = cancellationToken;
+        }
+
+        public override void SendUserMessage(PID pid, object message, PID sender)
+        {
+            if (message is T)
+            {
+                if (_cancellationToken.IsCancellationRequested) return;
```

Comment:
If cancellation is requested, Stop() is already called by the RequestAsync timeout continuation Task.
Could we call it twice ?
By the way, i'm not a big fan of this intricated code I made : Once stop() is called by the upper caller, once it's called by the FutureProcess itself, if you think of better design : what I need is a way to timeout IN the future.

Perhaps It's time to code an AlarmClock actor somewhere to send DelayMessages to any actor, with this the code of this timeout story could be nicer.